automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookTestUtils.scala

```diff
@@ -305,4 +308,35 @@ trait NotebookTestUtils extends LeonardoTestUtils {
       case other => fail(s"Unexpected kernel: $other")
     }
   }
+
+  def getLockedBy(workspaceBucketName: GcsBucketName, notebookName: GcsBlobName): IO[Option[String]] = {
+    google2StorageResource.use {
+      google2StorageDAO =>
+        for {
+          metadata <- google2StorageDAO.getObjectMetadata(workspaceBucketName, notebookName, None).compile.last
+          lockExpiresAt = metadata match {
+            case Some(GetMetadataResponse.Metadata(_, metadataMap)) if metadataMap.contains("lockExpiresAt") => Some(metadataMap("lockExpiresAt"))
+            case _ => None
+          }
+          currentlyLocked = lockExpiresAt match {
+            case Some(instantStr) => Instant.parse(instantStr).compareTo(Instant.now()) == 1
+            case None => false
+          }
+          lastLockedBy = if (currentlyLocked) {
+            metadata match {
+              case Some(GetMetadataResponse.Metadata(_, metadataMap)) if metadataMap.contains("lastLockedBy") => Some(metadataMap("lastLockedBy"))
+              case _ => None
+            }
+          } else None
+        } yield lastLockedBy
+    }
+  }
+
+  def getObject(workspaceBucketName: GcsBucketName, notebookName: GcsBlobName): IO[Option[String]] = {
```

Comment:
I found something that may be worth looking into: there's a class ContentItem and a function handleContentItemResponse that parses a JSON response from the jupyter server API into the class representation. Not sure if it's applicable to the response google gives, but it could be.